docs/rfc/active/000-ml_models/README.md

```diff
@@ -0,0 +1,123 @@
+
+## Detailed design
+![high level design](high_level_design.png)
+
+
+As shown in the above diagram, machine learning models are using machine learning features as inputs. These machine learning features
+could be shared across different machine learning models. In the example sketched above, `ML_Feature_1` and `ML_Feature_2` are used as inputs for `ML_Model_A` while `ML_Feature_2`, `ML_Feature_3` and `ML_Feature_4` are inputs for `ML_Model_B`.
+
+### URN Representation
+We'll define two [URNs](../../../what/urn.md): `MLModelUrn` and `MLFeatureUrn`.
+These URNs should allow for unique identification of machine learning models and features, respectively. Machine learning models, like datasets, will be identified by combination of standardized platform urn, name of the model and the fabric type where the model belongs to or where it was generated. 
+
+A machine learning model URN will look like below:
+```
+urn:li:mlModel:(<<platform>>,<<modelName>>,<<fabric>>)
+```
+A machine learning feature will be uniquely identified by it's name and the namespace this feature belongs to.
+A machine learning feature URN will look like below:
+```
+urn:li:mlFeature:(<<namespace>>,<<featureName>>)
+```
+
+### Entities
+There will be 2 top level GMA [entities](../../../what/entity.md) in the design: ML models and ML features.
+It's important to make ML features as a top level entity because ML features could be shared between different ML models.
+
+### ML Model metadata
+- Model properties: Basic information about the ML model
+  - Model date
+  - Model desription
+  - Model version
+  - Model type: Basic model architecture details e.g. if it is Naive Bayes classifier, Convolutional Neural Network, etc
+  - ML features used for training
+  - Hyperparameters of the model, used to control the learning process
+  - Tags: Primarily to enhance search and discovery of ML models
+- Ownership: Users who own the ML model, to help with directing questions or comments about the model.
+- Intended Use
+  - Primary intended use cases
+  - Primary intended user types
+  - Out-of-scope use cases
+- Model Factors: Factors affecting model performance including groups, instrumentation and environments
+  - Relevant Factors: Foreseeable factors for which model performance may vary
+  - Evaluation Factors: Factors that are being reported
+- Metrics: Measures of model performance being reported, as well as decision thresholds (if any) used.
+- Training Data: Details on datasets used for training ML Models
+  - Datasets used to train the ML model
+  - Motivation behind choosing these datasets
+  - Preprocessing steps involved: crucial for reproducibility
+  - Link to the process/job that captures training execution
```

Comment:
Same here, should we include the execution job link or the data pipeline code?

Reply:
Sure.